pkg/commands/start_standalone.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/rancher-sandbox/rancher-desktop-host-resolver/pkg/dns"
+	log "github.com/sirupsen/logrus"
+)
+
+func StartStandAloneServer(address string,
+	udpLocalPort,
+	tcpLocalPort int,
+	ipv6 bool,
+	hosts map[string]string,
+	upstreamServers []string) error {
+	var err error
+	if udpLocalPort == 0 {
+		udpLocalPort, err = randomUDPPort()
+		if err != nil {
+			return err
+		}
+	}
+	if tcpLocalPort == 0 {
+		tcpLocalPort, err = randomTCPPort()
+		if err != nil {
+			return err
+		}
+	}
+	srv, err := dns.Start(dns.ServerOptions{
+		Address:         address,
+		UDPPort:         udpLocalPort,
+		TCPPort:         tcpLocalPort,
+		IPv6:            ipv6,
+		StaticHosts:     hosts,
+		UpstreamServers: upstreamServers,
+	})
+	if err != nil {
+		return err
+	}
+	log.Infof("Started Stand Alone srv %+v", srv)
+	defer srv.Shutdown()
+
+	run()
+	return nil
+}
+
+func randomTCPPort() (int, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
+	defer l.Close()
+	return l.Addr().(*net.TCPAddr).Port, nil
```

Comment:
Does that actually work? Wouldn't the port be in `CLOSED_WAIT` and the server would be unable to listen on it (until some time out)?

Reply:
it works, I'm using it in all the tests.